### PR TITLE
Vibe 2.0 - css modules migration 1-3

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/3532714909/pulses/4318543772

- [ ] 1 - https://github.com/mondaycom/monday-ui-react-core/pull/1134
  - [ ] a - https://github.com/mondaycom/monday-ui-react-core/pull/823
    - [ ] 1. Tab
    - [ ] 2. TabPanel
    - [ ] 3. TextWithHighlight
    - [ ] 4. VirtualizedGrid
    - [ ] 5. VirtualizedList
  - [ ] b - https://github.com/mondaycom/monday-ui-react-core/pull/824
    - [ ] 6. TabsContext
    - [ ] 7. TabPanels
    - [ ] 8. TabList
    - [ ] 9. Tooltip
    - [ ] 10. Toast
    - [ ] 11. ToastLink

- [ ] 2 - https://github.com/mondaycom/monday-ui-react-core/pull/1161
  - [ ] a - https://github.com/mondaycom/monday-ui-react-core/pull/825
    - [ ] 1. RadioButton
    - [ ] 2. ResponsiveList
    - [ ] 3. Search
    - [ ] 4. Skeleton
    - [ ] 5. TextField
  - [ ] b - https://github.com/mondaycom/monday-ui-react-core/pull/826
    - [ ] 6. Icon
    - [ ] 7. List
    - [ ] 8. ListTitle
    - [ ] 9. ListItem
    - [ ] 10. ListItemIcon
    - [ ] 11. Link
    - [ ] 12. Label 

- [ ] 3 - https://github.com/mondaycom/monday-ui-react-core/pull/1166
  - [ ] a - https://github.com/mondaycom/monday-ui-react-core/pull/841
    - [ ] 1. EditableHeading
    - [ ] 2. EditableInput
    - [ ] 3. ExpandCollapse
    - [ ] 4. FieldLabel
    - [ ] 5. FormattedNumber
    - [ ] 6. Heading
    - [ ] 7. HiddenText
  - [ ] b - https://github.com/mondaycom/monday-ui-react-core/pull/842
    - [ ] 8. Counter
    - [ ] 9. Dialog
    - [ ] 10. DialogContent
    - [ ] 11. DialogContentContainer
    - [ ] 12. Divider
    - [ ] 13. Dropdown/menu
    - [ ] 14. Dropdown/option

4 -  [ON REVIEW] https://github.com/mondaycom/monday-ui-react-core/pull/1223
5 - [ON REVIEW] https://github.com/mondaycom/monday-ui-react-core/pull/1218

### CODE REVIEW

File: src/components/ResponsiveList/ResponsiveList.jsx [hunk 1]
```diff
@@ -135,4 +140,6 @@ ResponsiveList.defaultProps = {
   resizeDebounceTime: 0
 };
 
+ResponsiveList.defaultTestId = ComponentDefaultTestId.RESPONSIVE_LIST;
```
Comment (shlomitc) [inline src/components/ResponsiveList/ResponsiveList.jsx:143]: why is this needed? I saw it also in other components but I don't see why it's necessary

File: src/components/RadioButton/RadioButton.tsx [hunk 2]
```diff
@@ -126,4 +142,8 @@ const RadioButton: VibeComponent<RadioButtonProps, HTMLElement> = forwardRef(
   }
 );
 
+Object.assign(RadioButton, {
+  defaultTestId: ComponentDefaultTestId.RADIO_BUTTON
```
Comment (shlomitc) [inline src/components/RadioButton/RadioButton.tsx:146]: why is this needed?

File: src/components/Counter/__tests__/__snapshots__/counter-snapshot-tests.jest.tsx.snap [hunk 3]
```diff
@@ -176,13 +193,15 @@ exports[`Counter renders correctly with the prefix 1`] = `
 <span
   aria-label=" 13"
   aria-labelledby=""
+  className=""
```
Comment (shlomitc) [inline src/components/Counter/__tests__/__snapshots__/counter-snapshot-tests.jest.tsx.snap:196]: this should probably either not appear at all or contain something

File: src/components/AvatarGroup/__tests__/__snapshots__/avatarGroup-snapshot-tests.jest.js.snap [hunk 4]
```diff
@@ -323,13 +329,15 @@ exports[`AvatarGroup renders correctly renders correctly with custom counter too
     <span
       aria-label="Tab for more items 1"
       aria-labelledby=""
+      className=""
```
Comment (shlomitc) [inline src/components/AvatarGroup/__tests__/__snapshots__/avatarGroup-snapshot-tests.jest.js.snap:332]: same empty here

**Review by shlomitc — CHANGES_REQUESTED** [review]

File: src/components/ResponsiveList/ResponsiveList.jsx (hunk 1 above)
Comment (SergeyRoyt) [reply src/components/ResponsiveList/ResponsiveList.jsx:143]: Basically it serves two points:
1. Get default data-testid of the component without importing this big file with all the constants
2. See this defaul data-testid in the component Storybook props table

Comment (shlomitc) [reply src/components/ResponsiveList/ResponsiveList.jsx:143]: Why not simply export the default test ids file instead of re-exporting per component.
Also thinking about our cypress tests - I don’t want users to import jsx files - only raw js files

File: src/components/AvatarGroup/__tests__/__snapshots__/avatarGroup-snapshot-tests.jest.js.snap (hunk 4 above)
Comment (SergeyRoyt) [reply src/components/AvatarGroup/__tests__/__snapshots__/avatarGroup-snapshot-tests.jest.js.snap]: Removed

File: src/components/Counter/__tests__/__snapshots__/counter-snapshot-tests.jest.tsx.snap (hunk 3 above)
Comment (SergeyRoyt) [reply src/components/Counter/__tests__/__snapshots__/counter-snapshot-tests.jest.tsx.snap]: Removed

File: src/components/ResponsiveList/ResponsiveList.jsx (hunk 1 above)
Comment (SergeyRoyt) [reply src/components/ResponsiveList/ResponsiveList.jsx:143]: So currently we are exporting this file with constants in `interactionTests.js`
This defaultTestId is supposed to be a solution when user don't want to bring in whole `interactionTests.js`, just in order to get, let's say Clickable default data-testid - he'd be able just to get from `Clickable.defaultTestId`

Comment (shlomitc) [reply src/components/ResponsiveList/ResponsiveList.jsx:143]: I think it’s better to have one way to use it instead of multiple ones and the single file is better in terms of cross-platform testing

Comment (SergeyRoyt) [reply src/components/ResponsiveList/ResponsiveList.jsx:143]: Removed in Vibe 2.0 https://github.com/mondaycom/monday-ui-react-core/pull/1254
